docs/en/join-testnet.md

```diff
@@ -4,58 +4,168 @@ title: Join Testnets
+
+## Requirements
+
+Make sure the following requirements are met and then follow the steps in the next section:
+
+```text
+Node >= 8
+yarn or npm
+```
+
+## Setting Things Up
+
+First, let's clone the [Truffle DAppChainExample](https://github.com/loomnetwork/truffle-dappchain-example/) repository. Open a terminal, `cd` into the directory where you store your projects, and run the following command:
+
+```bash
+git clone https://github.com/loomnetwork/truffle-dappchain-example
+```
+
+Next, `cd` into `truffle-dappchain-example`:
+
+```bash
+cd truffle-dappchain-example
+```
+
+and run:
+
+```bash
+yarn install
+```
+
+## Run against Loom Testnet
+
+### Generate the Loom private key
+
+The following command will download the `loom` binary and generate the private key for you:
+
+```bash
+yarn gen:extdev-key
+```
+
+> The Loom private key will be saved into a file called `extdev_private_key`.
+
+
+### Deploy to extdev_plasma_us1
+
+As an example, we will deploy and run our application against `extdev_plasma_us1`.
+
+Run the following command:
+
+```bash
+yarn deploy:extdev
+```
+
+>The above command will **compile and then deploy** our smart contract to `extdev_plasma_us1`
+
+
+## Testing
+
+Run this command below to send transactions to the smart contract deployed to `extdev_plasma_us1`:
+
+
+```test
+yarn test:extdev
+```
+
+If everything looks good, let's spin up a web server and interact with our smart contract.
+
+## Start the Web Interface
+
+The web interface is built with React and Webpack. Open a new terminal and run the following command:
+
+```bash
+yarn serve:extdev
+```
+
+> The web interface is available on http://localhost:8080.
+
+
+## Run Against Local Loom Chain
+
+First, you have to generate a private key using:
+
+```bash
+yarn gen:loom-key
+```
+
+### Spin up Loom Chain
+
+In a new terminal, run:
 
 ```bash
-./loom genkey -k priv_key -a pub_key
+yarn loom-init
 ```
 
-You should see something like this printed out in the console:
+### Deploy the Smart Contract
+
+To deploy, run the following command:
+
+```bash
+yarn deploy
+```
+
+You will see something like this:
 
 ```text
-local address: 0x3B334bEd1e7d3e7d9214495120160D9236aCbC31
-local address base64: OzNL7R59Pn2SFElRIBYNkjasvDE=
+Starting migrations...
+======================
+> Network name:    'loom_dapp_chain'
+> Network id:      13654820909954
+> Block gas limit: 0
+```
+
+Copy the `Network id` paste it into a text file. We'll use it later.
+
+### Testing
+
+To test it, just run the following command:
+
+```bash
+yarn test
 ```
 
-This will create two files named `priv_key` and `pub_key`. The `priv_key` file contains your private key that you'll use to deploy contracts to the DAppChain.
+### Web interface
 
+Open the `src/contract.js` file in your text editor of choice and go to line 87.
 
-### Deploy Truffle Example to a Testnet
+Now, say your `Network id` is `13654820909954`. Replace
```

Comment:
Why is this needed ?

Reply:
It tries to connect with `default` but this doesn't work. It just errors out so I added a comment to the source code and this paragraph to the docs. Can't remember if I tried with `*` instead of `default`. I'll give it a shot.

https://github.com/loomnetwork/truffle-dappchain-example/blob/0ee45592908e32468f96d0f7a406b190dddb98a0/src/contract.js#L87

Reply:
Tried with `*`... it doesn't work either. Maybe I should dig into this a bit more and figure out a better way of doing it.